modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -196,15 +242,48 @@ public static HttpRequestBase buildHeaders(HttpRequestBase httpRequestBase,
 		} catch (IOException e) {
 			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
 			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
-			                   Constants.INTERNAL_SERVER_ERROR);
+					Constants.INTERNAL_SERVER_ERROR);
 			throw new IDPTokenManagerException("Server connectivity failure.", e);
 		}
-		
+
+		return responseParams;
+	}
+
+	public static Map<String, String> sendPutRequest(String url, String params, Map<String,
+			String> headers) throws IDPTokenManagerException {
+
+		HttpPut httpPut = new HttpPut(url);
+		HttpClient httpClient = getCertifiedHttpClient();
+		HttpPut httpPutWithHeaders = (HttpPut) buildHeaders(httpPut, headers);
+		Map<String, String> responseParams = new HashMap<String, String>();
+
+		if (params != null) {
+			try {
+				httpPutWithHeaders.setEntity(new StringEntity(params));
+			} catch (UnsupportedEncodingException e) {
+				throw new IDPTokenManagerException("Invalid encoding type.", e);
+			}
+		}
+		try {
+			HttpResponse response = httpClient.execute(httpPutWithHeaders);
+			String status = String.valueOf(response.getStatusLine().getStatusCode());
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS, status);
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+
+		} catch (ClientProtocolException e) {
+			throw new IDPTokenManagerException("Invalid client protocol.", e);
```

Comment:
Do we not need to add the two parameters configured in the next catch block, here as well?